arctic/store/_ndarray_store.py

```diff
@@ -268,14 +292,21 @@ def _do_read(self, collection, version, symbol, index_range=None):
         segment_count = None
 
         spec = {'symbol': symbol,
-                'parent': version_base_or_id(version),
                 'segment': {'$lt': to_index}
                 }
         if from_index is not None:
             spec['segment']['$gte'] = from_index
         else:
             segment_count = version.get('segment_count', None)
 
+        # We want to use FW pointers to read data if:
+        #  ARCTIC_FORWARD_POINTERS is currently enabled/hybrid AND last version was written with FW pointers,
```

Comment:
Should we consider making enabled use the reverse pointers to read, but then checking that the forward pointers reconcile?  As a way of validating for now?

Just an idea, I'm not against this approach, just asking what we want to test during the hybrid mode period.

Reply:
The written data are being verified upon write, i.e. we count the segments matching the write spec which have been actually written.
https://github.com/dimosped/arctic/blob/forward_pointers_final/arctic/store/_ndarray_store.py#L505

Tbh I am not against this idea, but maybe as a separate option/flag and add it as part of the verification check:
https://github.com/dimosped/arctic/blob/forward_pointers_final/arctic/store/_ndarray_store.py#L505

It would also mean r/w will be slower with an extra step of  1-query collecting N-document IDs.

Reply:
I actually went ahead and implemented this, but during testing of forcing a failed check, a nasty bug was uncovered.

If mongo_retry operation fails once, the successfully written segments always remain (it is Arctic's design decision which won't change now). The upserted ids though in subsequent retried, don't include them, breaking the model. 

Working on it and will add explicit tests.